inc/ticket.class.php

```diff
@@ -6288,11 +6288,19 @@ function showTimeline($rand) {
          $options = array( 'parent' => $this,
                            'rand' => $rand
                            ) ;
-         $obj = new $item['type'] ;
-         $obj->fields = $item['item'] ;
+         $obj = getItemForItemtype($item['type']);
+         if( $obj ){
+            $obj->fields = $item['item'] ;
+         } else {
+            $obj = $item ;
+         }
          Plugin::doHook('pre_show_item', array('item' => $obj, 'options' => &$options));
 
-         $item_i = $obj->fields;
+         if( is_array( $obj ) ){
+            $item_i = $obj['item'];
+         } else {
+            $item_i = $obj->fields;
+         }
 
```

Comment:
This part seems very complex, initially it was (before 94c8fada5438fa35126cb97f213fd0bf8476a3ef) : 
```php
$item_i = $item['item'];
```

Why redefining this var as it's not used by the new hooks ?